keystorage/src/main/java/tech/pegasys/web3signer/keystorage/awskms/AwsKeyManagerService.java

```diff
@@ -0,0 +1,172 @@
+import software.amazon.awssdk.services.kms.model.KeySpec;
+import software.amazon.awssdk.services.kms.model.MessageType;
+import software.amazon.awssdk.services.kms.model.SignRequest;
+import software.amazon.awssdk.services.kms.model.SignResponse;
+import software.amazon.awssdk.services.kms.model.SigningAlgorithmSpec;
+
+/**
+ * Wraps AWS KMS Client and expose public key and sign operations. Since KmsClient is thread safe,
+ * same instance of this class can be used to talk to same host/credentials.
+ */
+public class AwsKeyManagerService implements AutoCloseable {
+  private final AwsAuthenticationMode authMode;
+  private final AwsCredentials awsCredentials;
+  private final String region;
+
+  private KmsClient kmsClient;
+
+  private Optional<URI> endpointOverride;
+
+  public AwsKeyManagerService(
+      final AwsAuthenticationMode authMode,
+      final AwsCredentials awsCredentials,
+      final String region,
+      final Optional<URI> endpointOverride) {
+    this.authMode = authMode;
+    this.awsCredentials = awsCredentials;
+    this.region = region;
+    this.endpointOverride = endpointOverride;
+
+    initKmsClient();
+  }
+
+  void initKmsClient() {
+    if (kmsClient != null) {
+      return;
+    }
+
+    final KmsClientBuilder kmsClientBuilder = KmsClient.builder();
+    endpointOverride.ifPresent(kmsClientBuilder::endpointOverride);
+    kmsClient =
+        kmsClientBuilder
+            .credentialsProvider(getAwsCredentialsProvider())
+            .region(Region.of(region))
+            .build();
+  }
+
+  private AwsCredentialsProvider getAwsCredentialsProvider() {
+    final AwsCredentialsProvider awsCredentialsProvider;
+    if (authMode == ENVIRONMENT) {
+      awsCredentialsProvider = DefaultCredentialsProvider.create();
+    } else {
+      final software.amazon.awssdk.auth.credentials.AwsCredentials awsSdkCredentials;
+      if (awsCredentials.getSessionToken().isPresent()) {
+        awsSdkCredentials =
+            AwsSessionCredentials.create(
+                awsCredentials.getAccessKeyId(),
+                awsCredentials.getSecretAccessKey(),
+                awsCredentials.getSessionToken().get());
+      } else {
+        awsSdkCredentials =
+            AwsBasicCredentials.create(
+                awsCredentials.getAccessKeyId(), awsCredentials.getSecretAccessKey());
+      }
+
+      awsCredentialsProvider = StaticCredentialsProvider.create(awsSdkCredentials);
+    }
+    return awsCredentialsProvider;
+  }
+
+  public ECPublicKey getECPublicKey(final String kmsKeyId) {
+    checkArgument(kmsClient != null, "KmsClient is not initialized");
+
+    // Question ... do we need to set grantTokens?
+    final GetPublicKeyRequest getPublicKeyRequest =
+        GetPublicKeyRequest.builder().keyId(kmsKeyId).build();
+    final GetPublicKeyResponse publicKeyResponse = kmsClient.getPublicKey(getPublicKeyRequest);
+    KeySpec keySpec = publicKeyResponse.keySpec();
+    if (keySpec != KeySpec.ECC_SECG_P256_K1) {
```

Comment:
Our Azure code uses two SignatureAlgorithms (one deprecated) - do we need to worry about that for AWS or is it an Azure specific thing?